docs/next/guides/building-and-testing/plugins.md

```diff
@@ -17,228 +15,242 @@ tags:
 
 ## Overview
 
-Plugin is a one great way to extend the capabilities of Handsontable. In fact, most of the features available in this library are provided by plugins.
+Plugin is a one great way to extend the capabilities of Handsontable. In fact, most of features available in this library are provided by plugins.
 
-## Plugin template
+This guide will lead you, step-by-step, through the process of creating a custom extension.
 
-There are two types of plugins: internal and external. While both extend Handsontable's functionality, the former is incorporated into the Handsontable build, and the latter needs to be included from a separate file.
+### 1. Prerequisites
 
-Regardless of which plugin you are going to build, using a template will save you lots of time.
+You need to import the following:
+- `BasePlugin` - contains the basic interface to work within Handsontable's lifecycle,
+- `registerPlugin` - utility to register the plugin in Handsontable plugins registry.
 
-### Internal plugin
+Both provide the basic interface necessary to create a reusable plugin.
+
+```js
+import { BasePlugin, registerPlugin } from 'handsontable/plugins';
+```
+
+### 2. Extend BasePlugin
+Extending our `BasePlugin` is a recommended way to start creating your next extension.
+In this built-in interface, we care about many factors such as backward compatibility, memory leaks preventions and proper binding plugin's instance with Handsontable.
 
-The source code of this template is [available on GitHub](https://github.com/handsontable/handsontable-skeleton/tree/master/plugins/internal). Start off by cloning the entire project or copying the code to your application.
 
 ```js
-// You need to import the BasePlugin class in order to inherit from it.
-import BasePlugin from './../../_base';
-import {registerPlugin} from './../../../plugins';
-
-/**
- * @plugin InternalPluginSkeleton
- * Note: keep in mind, that Handsontable instance creates one instance of the plugin class.
- *
- * @description
- * Blank plugin template. It needs to inherit from the BasePlugin class.
- */
-class InternalPluginSkeleton extends BasePlugin {
-
-  // The argument passed to the constructor is the currently processed Handsontable instance object.
+export class CustomPlugin extends BasePlugin {
+  /**
+   * Define a unique plugin's key.
+   * Handsontable uses that string to register the plugin under that alias.
+   * Moreover, you might also use the same parameter to recognize the plugin's
+   * options passed through the Setting object at Handsontable initialization.
+   *
+   * @returns {string}
+   */
```

Comment:
/**
  \* Define a unique key (a string) for your plugin.
   \* The key becomes the plugin's alias.
   \* Handsontable registers the plugin under that alias.
   \* You can also use the alias to recognize the plugin's
   \* options passed through the Setting object
   \* at Handsontable's initialization.
   \*
   \* @returns {string}
   \*/